nrf-hal-common/src/saadc.rs

```diff
@@ -1,4 +1,20 @@
 //! HAL interface to the SAADC peripheral.
+//!
+//! example usage:
+//! ```rust
+//! use nrf52840_hal::prelude::*;
+//! use nrf52840_hal::saadc::SaadcConfig as SaadcConfig;
+//! use nrf52840_hal::saadc::Saadc as Saadc;
+//!
+//! let mut saadc_pin = /* the pin your analog device is connected to */;
```

Comment:
This likely won't compile when tested. When you run `cargo test`, all doc examples will be compiled and ran. It's also more idiomatic to use an actual example (pick one of the ADC pins, like p0_03).

Reply:
Hrm. If I run `cargo test` or `cargo test --doc` in the `nrf52840-hal`subdirectory, 0 tests run. In the root it won't compile obviously. How do I try this out?
(I'm asking because I want to figure out which parts I can omit and wich I can't (since I've seen many examples that have imports but no function wrapped around the code), and how imports would work)

Reply:
note to future self: it was `➜  nrf-hal-common git:(saadc-docs) ✗ cargo test --doc  --features=52840`